src/python/pants/backend/python/dependency_inference/scripts/import_parser.py

```diff
@@ -26,6 +26,12 @@
     re.UNICODE,
 )
 
+def _consume_until(predicate, iterable):
+    """Returns the first element in iterable where predicate is True."""
+    for element in iterable:
+        if predicate(element):
+            return element
```

Comment:
I'm a little confused how this works. Using `return` will break out of the whole function so you would only ever return at most one thing. I'd think you'd want to return multiple things using `yield`?

Perhaps this is mutating the iterator, such that it gets called again on the same iterator but with a different location?

It'd be hepful to add type hints to this, maybe explain in a comment how it works. I think `typing.Iterator`? Not iterable. If so, renaming the variable to `iterator` would help too.

Reply:
We can't add type annotations pythonically because of the Py2 issue :roll_eyes: 

I think this can be replaced by `itertools.dropwhile`.

Reply:
Oh yeah! Bummer. K, thanks :)

Reply:
Pex would beg to differ. Do type comments not work?

Reply:
They do, I'm hiding behind "pythonically" because type comments are quite a pain :grimacing: